src/hooks/useSamplesContext.js

```diff
@@ -0,0 +1,130 @@
+import { useContext, useEffect, useRef, useState } from 'react'
+import { api } from 'api'
+import { SamplesContext } from 'contexts/SamplesContext'
+
+export const useSamplesContext = (initialQueries) => {
+  const { samplesQuery, setSamplesQuery } = useContext(SamplesContext)
+  const prevQueryRef = useRef(samplesQuery) // previous samplesQuery reference
+  const [loading, setLoading] = useState(false)
+  const [hasError, setHasError] = useState(false)
+  const [samples, setSamples] = useState([])
+  const hasSamples = samples?.results?.length > 0
+  const totalSamples = (samples && samples.count) || 0
+
+  // initial fetch on page load
+  useEffect(() => {
+    getSamples()
+  }, [])
+
+  // refetches on samplesQuery changes
+  useEffect(() => {
+    if (isQueryChanged) {
+      prevQueryRef.current = samplesQuery
+      getSamples()
+    }
+  }, [samplesQuery])
+
+  /* Common */
+  const getSamples = async () => {
+    const { offset, limit, datasetId, filterBy, ordering } = samplesQuery
+    const params = {
+      ...initialQueries,
+      offset,
+      limit,
+      ...(datasetId && { dataset_id: datasetId }),
+      ...(filterBy && { filter_by: filterBy }),
```

Comment:
No need to rename these since they are for the API and inside an object so we should just spread samplesQuery with initialQuery into a new object.

Reply:
Here, to avoid using the logical AND `...(propName && { ...`, I also added a new `filterNullAndEmptyFromObject` helper to remove `null` and empty string values from `params` and spread `samplesQuery` directly.